src/main/java/hudson/plugins/parameterizedtrigger/TriggerBuilder.java

```diff
@@ -134,15 +134,19 @@ public boolean perform(AbstractBuild<?, ?> build, Launcher launcher,
                         }
                         for (Future<Run> future : futures.get(p)) {
                             try {
-                                listener.getLogger().println("Waiting for the completion of " + HyperlinkNote.encodeTo('/'+ p.getUrl(), p.getFullDisplayName()));
-                                Run b = future.get();
-                                listener.getLogger().println(HyperlinkNote.encodeTo('/'+ b.getUrl(), b.getFullDisplayName()) + " completed. Result was "+b.getResult());
-                                BuildInfoExporterAction.addBuildInfoExporterAction(build, b.getParent().getFullName(), b.getNumber(), b.getResult());
-
-                                if(buildStepResult && config.getBlock().mapBuildStepResult(b.getResult())) {
-                                    build.setResult(config.getBlock().mapBuildResult(b.getResult()));
+                                if (future != null ) {
+                                    listener.getLogger().println("Waiting for the completion of " + HyperlinkNote.encodeTo('/'+ p.getUrl(), p.getFullDisplayName()));
+                                    Run b = future.get();
+                                    listener.getLogger().println(HyperlinkNote.encodeTo('/' + b.getUrl(), b.getFullDisplayName()) + " completed. Result was " + b.getResult());
+                                    BuildInfoExporterAction.addBuildInfoExporterAction(build, b.getParent().getFullName(), b.getNumber(), b.getResult());
+
+                                    if (buildStepResult && config.getBlock().mapBuildStepResult(b.getResult())) {
+                                        build.setResult(config.getBlock().mapBuildResult(b.getResult()));
+                                    } else {
+                                        buildStepResult = false;
+                                    }
                                 } else {
-                                    buildStepResult = false;
+                                    listener.getLogger().println("Skipping " + HyperlinkNote.encodeTo('/'+ p.getUrl(), p.getFullDisplayName()) + ". The project was disabled.");
```

Comment:
Several cases:
- perform3 Javadoc: `// When a project is disabled or the configuration is not yet saved f will always be null and we're caught in a loop, therefore we need to check for it`
- BuildTriggerConfig#schedule:
 * job is not an instance of `ParameterizedJobMixIn.ParameterizedJob`
* `ParameterizedJobMixIn#schedule2()` returns null due to whatever other reason: Job is disabled, job is not buildable (e.g. triggering of Matrix Configuration), or Jenkins just does not like you (dispatcher extensions return null) :(